src/h5web/vis-packs/core/heatmap/config.tsx

```diff
@@ -99,7 +100,14 @@ export function useHeatmapConfig<U>(
 // https://github.com/pmndrs/zustand/issues/128#issuecomment-673398578
 export function HeatmapConfigProvider(props: ConfigProviderProps) {
   const { children } = props;
-  const [useStore] = useState(initialiseStore); // https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
 
-  return <Provider value={useStore}>{children}</Provider>;
+  // https://github.com/pmndrs/zustand/issues/346
+  const [reydrated, setRehydrated] = useState(false);
+
+  // https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
+  const [useStore] = useState(() => {
+    return initialiseStore(() => setRehydrated(true));
+  });
+
+  return reydrated ? <Provider value={useStore}>{children}</Provider> : null;
```

Comment:
I'm watching the relevant issue. Hopefully zustand will provide an out-of-the-box solution soon.